src/plugins/Workers/SyncWorker/initialSyncUp.js

```diff
@@ -10,23 +11,28 @@ const fetchAndStoreAddressTransactions = require('./utils/fetchAndStoreAddressTr
 // fetching here found that we have more addresses that on starting of the sync up.
 module.exports = async function initialSyncUp() {
   const { transporter, storage } = this;
-
   const addrList = this.getAddressListToSync().map((addr) => addr.address);
 
   // Due to the events system, we need to handle the fact that we did subscribed to addresses
   // that we had received the transactions and store before
   // being able to release initialSyncUp as ready.
   // When we will move to bloomfilter, that part might be more complex.
-  /* eslint-disable-next-line no-async-promise-executor */
-  return new Promise(async (resolve) => {
-    // eslint-disable-next-line no-restricted-syntax
-    for (const address of addrList) {
-      // We need to wait for fetching and addition to be done sequentially
-      // to correctly manage our balance as input can be output in prev tx.
-      /* eslint-disable-next-line no-await-in-loop */
-      await fetchAndStoreAddressTransactions(address, transporter, storage);
-    }
-    logger.silly('SyncWorker - initialSyncUp - Fully synced');
-    resolve();
-  });
+
+
+  const transactionPromises = addrList.map(_.bind(fetchAddressTransactions, null, _, transporter));
+
+  const transactionsByAddresses = await Promise.all(transactionPromises);
+
+  const transactions = _.flatten(transactionsByAddresses);
+
+  const ordered = new TransactionOrderer();
+
+  transactions.forEach((tx) => ordered.insert(tx));
+
+  const importPromises = ordered.transactions
```

Comment:
The same, could we put the function into importTransactionToStorage variable and then pass it to map